### PR TITLE
Add storage class related tests and example

### PR DESCRIPTION
**Pull request by nitisht**

### CODE REVIEW

**Comment by nitisht** [comment]
Play server needs to be updated to make sure the tests pass.

**Comment by harshavardhana** [comment]
Play server is updated minio-go tests pass on play.minio.io

**Comment by harshavardhana** [comment]
The Travis failure is in checkStyle

**Comment by nitisht** [comment]
> The Travis failure is in checkStyle

oh..fixed checkStyle now

File: examples/PutObjectWithMetadata.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        builder.append("A few others we like: Amazingly few discotheques provide jukeboxes; Now fax quiz Jack! my ");
+        builder.append("brave ghost pled; Watch Jeopardy!, Alex Trebeks fun TV quiz game.\n");
+        builder.append("---\n");
+      }
+
+      // Create a InputStream for object upload.
+      ByteArrayInputStream bais = new ByteArrayInputStream(builder.toString().getBytes("UTF-8"));
+
+      Map<String, String> headerMap = new HashMap<>();
+      headerMap.put("X-Amz-Storage-Class", "REDUCED_REDUNDANCY");
```
Comment (balamurugana) [inline examples/PutObjectWithMetadata.java]: Why do we need server specific examples in SDKs?

Comment (balamurugana) [reply examples/PutObjectWithMetadata.java]: Add more headers to have it complete how users really use them.

Comment (nitisht) [reply examples/PutObjectWithMetadata.java]: sure will do

Comment (nitisht) [reply examples/PutObjectWithMetadata.java]: Added other metadata fields here @balamurugana

**Comment by harshavardhana** [comment]
looks like the tests are not running yet @nitisht

**Comment by balamurugana** [comment]
Travis fails with gradle/java9 incompatibility.

**Pull request merged**